src/main/connect/connectionManager.ts

```diff
@@ -184,32 +177,34 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
     public areCompleteCredentialsSet(): boolean {
         return !!(this._xrayUrl && this._rtUrl && ((this._username && this._password) || this._accessToken));
     }
-
-    public async verifyCredentials(prompt: boolean): Promise<boolean> {
-        if (!this.areXrayCredentialsSet()) {
-            return false;
+    /**
+     * Test whether the connection is valid and the user has permissions to access Xray and Artifactory.
+     */
+    public async testConnection(prompt: boolean): Promise<TestConnectionStatus> {
+        if (!(await this.testConnectionHealth())) {
+            this.deleteCredentialsFromMemory();
+            return TestConnectionStatus.FailedServerNotFound;
         }
         if (
-            !(await ConnectionUtils.checkXrayConnectionAndPermissions(
-                this._xrayUrl,
-                this._username,
-                this._password,
-                this._accessToken,
-                prompt,
-                this._logManager
-            ))
+            !(await ConnectionUtils.checkXrayPermissions(this._xrayUrl, this._username, this._password, this._accessToken, prompt, this._logManager))
         ) {
-            return false;
+            this.deleteCredentialsFromMemory();
+            return TestConnectionStatus.FailedBadCredentials;
         }
-        if (this.areCompleteCredentialsSet()) {
-            return ConnectionUtils.checkArtifactoryConnection(
-                this._rtUrl,
-                this._username,
-                this._password,
-                this._accessToken,
-                prompt,
-                this._logManager
-            );
+        return TestConnectionStatus.Success;
+    }
+
+    public async testConnectionHealth(): Promise<boolean> {
+        try {
+            if (!(await ConnectionUtils.pingXray(this.xrayUrl, this._username, this._password, this._accessToken))) {
+                return false;
+            }
+            if (this.areCompleteCredentialsSet()) {
+                return await ConnectionUtils.pingArtifactory(this.rtUrl, this._username, this._password, this._accessToken);
+            }
+        } catch (error) {
+            this._logManager.logMessage('testConnectionHealth failed. Error:' + error, 'WARN');
```

Comment:
```suggestion
            this._logManager.logMessage('Test connection failed. Error: ' + error, 'WARN');
```
Also - there is a chance that the error message will be in error.message. See the `logError` function in logManager.ts. Maybe it worth using it instead.